### ISSUE TITLE
Can you share the yolov5x that finally used? And the script to transfer yolov5 official weights to TPH-YOLOv5?

### ISSUE THREAD

**Issue by clw5180** [cv516Buaa/tph-yolov5#13]
Thanks a lot!

**Comment by clw5180** [comment]
训练的时候，--weights yolov5l.pt 是不是直接会把yolov5官方模型里面有的key加载进来

**Comment by cv516Buaa** [comment]
We will not share the final yolov5x weights. The weights of yolov5l have proved the excellent performance of the tph-yolov5. You can train the weights of yolov5x by yourself.
If you want to use any pre-trained weights, use --weights 'path to your pretrained weights'.
For more information you can refer to the original repo of yolov5.

**Comment by phanisai22** [comment]
> 训练的时候，--weights yolov5l.pt 是不是直接会把yolov5官方模型里面有的key加载进来

im getting this error when I used these weights

!python tph-yolov5/val.py --weights yolov5l.pt --img 1996 --data tph-yolov5/data/VisDrone.yaml --augment --save-txt  --save-conf --task val --batch-size 8 --verbose --name v5l

![image](https://user-images.githubusercontent.com/42872894/147030761-eb3881af-d099-4211-9393-86030b0fd502.png)


> We will not share the final yolov5x weights. The weights of yolov5l have proved the excellent performance of the tph-yolov5. You can train the weights of yolov5x by yourself. If you want to use any pre-trained weights, use --weights 'path to your pretrained weights'. For more information you can refer to the original repo of yolov5.

**Comment by bilzard** [comment]
@cv516Buaa Can you share model definition of `yolov5x-xs-tph.yaml`?
Is just fixing `depth_multiple=1.33` and `width_multiple=1.25` in the `yolov5l-xs-tph.yaml` O.K.?

**Comment by bilzard** [comment]
> Is just fixing depth_multiple=1.33 and width_multiple=1.25 in the yolov5l-xs-tph.yaml O.K.?

Seems O.K.

I found this comment.
https://github.com/cv516Buaa/tph-yolov5/issues/6#issuecomment-983842916

**Comment by mx2013713828** [comment]
> > Is just fixing depth_multiple=1.33 and width_multiple=1.25 in the yolov5l-xs-tph.yaml O.K.?
> 
> Seems O.K.
> 
> I found this comment. [#6 (comment)](https://github.com/cv516Buaa/tph-yolov5/issues/6#issuecomment-983842916)

Hi,how about your experiment？
I also want yolov5s with TPH，how should I change the yaml？

**Comment by bilzard** [comment]
@mx2013713828 Check the diff of yolov5s.yaml and yolov5l.yaml. These are only differed with few lines.

**Comment by mx2013713828** [comment]
> @mx2013713828 Check the diff of yolov5s.yaml and yolov5l.yaml. These are only differed with few lines.

okey,I‘m training yolov5s，but my PC is not strong,so I'm waiting for result! can you share your experiment result with yolov5m?
thank you!

**Comment by bilzard** [comment]
Unfortunately currently I have no plan to train full scratch.

**Comment by mx2013713828** [comment]
> Unfortunately currently I have no plan to train full scratch.

Anyway,thanks!

**Comment by maarten0912** [comment]
Hello @mx2013713828 , I have trained the model from scratch with yolov5s and yolov5s6. For the yaml file I only changed the width and depth multipliers corresponding to the ones from the ultralytics repo. These were my results (pretty good!)

![image](https://user-images.githubusercontent.com/57093022/149747847-c89543f9-1dbe-48ae-aa38-641010a0dc09.png)

Note that `v5s6-xs-tph` is trained with input size 640 and `v5s6-xs-tph-1280` is trained with input size 1280. I don't really understand exactly if that is the way to do it, but this works pretty good.

**Comment by bilzard** [comment]
@maarten0912 Cool. Thank you for sharing. Can you also share mAP@.5:.95 plot?

**Comment by maarten0912** [comment]
@bilzard 
![image](https://user-images.githubusercontent.com/57093022/149890627-2973e413-b4c8-4878-90b4-719d85e59f9a.png)

**Comment by bilzard** [comment]
@maarten0912 Thanks!

**Comment by RoyCopterPix** [comment]
@maarten0912 can you share your yolov5s and yolov5s6 pretrained weights?  thanks!

**Comment by maarten0912** [comment]
> @maarten0912 can you share your yolov5s and yolov5s6 pretrained weights? thanks!

@RoyCopterPix They are both found here: https://github.com/ultralytics/yolov5#pretrained-checkpoints

**Comment by RoyCopterPix** [comment]
> > @maarten0912 can you share your yolov5s and yolov5s6 pretrained weights? thanks!
> 
> @RoyCopterPix They are both found here: https://github.com/ultralytics/yolov5#pretrained-checkpoints

I meant your pretraied weights with the tph-yolov5

**Comment by maarten0912** [comment]
> I meant your pretraied weights with the tph-yolov5

@RoyCopterPix  Yes, I used the yolov5s (trained on COCO) as pretrained weights for training the tph-yolov5 model. I trained that model on the VisDrone dataset. Do you maybe mean my final weights?

**Comment by RoyCopterPix** [comment]
> > I meant your pretraied weights with the tph-yolov5
> 
> @RoyCopterPix Yes, I used the yolov5s (trained on COCO) as pretrained weights for training the tph-yolov5 model. I trained that model on the VisDrone dataset. Do you maybe mean my final weights?

yeah I meant your final weights :) can you share it?

**Comment by maarten0912** [comment]
> > > I meant your pretraied weights with the tph-yolov5
> > 
> > 
> > @RoyCopterPix Yes, I used the yolov5s (trained on COCO) as pretrained weights for training the tph-yolov5 model. I trained that model on the VisDrone dataset. Do you maybe mean my final weights?
> 
> yeah I meant your final weights :) can you share it?

@RoyCopterPix I'm sorry, but I cannot share it. You can train your own model by modifying the .yaml, using the yolov5s as pretrained and training on the VisDrone dataset. You just have to follow the README. If you don't have a CUDA machine, you can rent a cloud machine.

**Comment by RoyCopterPix** [comment]
> > > > I meant your pretraied weights with the tph-yolov5
> > > 
> > > 
> > > @RoyCopterPix Yes, I used the yolov5s (trained on COCO) as pretrained weights for training the tph-yolov5 model. I trained that model on the VisDrone dataset. Do you maybe mean my final weights?
> > 
> > 
> > yeah I meant your final weights :) can you share it?
> 
> @RoyCopterPix I'm sorry, but I cannot share it. You can train your own model by modifying the .yaml, using the yolov5s as pretrained and training on the VisDrone dataset. You just have to follow the README. If you don't have a CUDA machine, you can rent a cloud machine.

Oh ok then I got 3-4 questions:

1. If I want to train the TPH-YOLOv5, I will need the original pretrained weights of the original yolov5 repo together with the VisDrone dataset? I can use any weights from the original yolov5, right?
2. Can you tell me if you get better results using the yolov5s6-xs-tph-1280 rather than yolov5l-xs-tph? (small 1280 against 640 Large, which the authors of the repo provide)
3. Do you think I will get extremely better results(detections) using yolov5m6-xs-tph-1280/yolov5l7-xs-tph-1280(after training) rather than the yolov5l-tph? 
4. How much time it took you to train the yolov5s6 on the visdrone dataset to get the yolov5s6-tph-1280 weights? 

Thanks!

**Comment by Zephyr69** [comment]
You could just create the yaml using that pic as the reference, and then train from scratch. On my custom tiny-target dataset, that final model performed worse than the other custom model provided by this repo(not the tph one, that one was the worst :/).

**Comment by mx2013713828** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
